receiver/azureeventhubreceiver/data_converter_impl.go

```diff
@@ -0,0 +1,232 @@
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"regexp"
+	"strconv"
+
+	"github.com/relvacode/iso8601"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+var notHexDigit = regexp.MustCompile(`[^a-fA-F0-9]+`)
+
+// AzureRecords represents an array of Azure log records
+// as exported via an Azure Event Hub
+type AzureRecords struct {
+	Records []AzureLogRecord `json:"records"`
+}
+
+// AzureLogRecord represents a single Azure log following
+// the common schema:
+// https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/resource-logs-schema
+type AzureLogRecord struct {
+	Time              string       `json:"time"`
+	ResourceID        string       `json:"resourceId"`
+	TenantID          *string      `json:"tenantId"`
+	OperationName     string       `json:"operationName"`
+	OperationVersion  *string      `json:"operationVersion"`
+	Category          string       `json:"category"`
+	ResultType        *string      `json:"resultType"`
+	ResultSignature   *string      `json:"resultSignature"`
+	ResultDescription *string      `json:"resultDescription"`
+	DurationMs        *string      `json:"durationMs"`
+	CallerIPAddress   *string      `json:"callerIpAddress"`
+	CorrelationID     *string      `json:"correlationId"`
+	Identity          *interface{} `json:"identity"`
+	Level             *string      `json:"Level"`
+	Location          *string      `json:"location"`
+	Properties        *interface{} `json:"properties"`
+}
+
+// AsTimestamp will parse an ISO8601 string into an OpenTelemetry
+// nanosecond timestamp. If the string cannot be parsed, it will
+// return zero and the error.
+func AsTimestamp(s string) (pcommon.Timestamp, error) {
+	t, err := iso8601.ParseString(s)
+	if err != nil {
+		return 0, err
+	}
+	return pcommon.Timestamp(t.UnixNano()), nil
+}
+
+// AsSeverity converts the Azure log level to equivalent
+// OpenTelemetry severity numbers. If the log level is not
+// valid, then the 'Unspecified' value is returned.
+func AsSeverity(s string) plog.SeverityNumber {
+	switch s {
+	case "Informational":
+		return plog.SeverityNumberInfo
+	case "Warning":
+		return plog.SeverityNumberWarn
+	case "Error":
+		return plog.SeverityNumberError
+	case "Critical":
+		return plog.SeverityNumberFatal
+	default:
+		return plog.SeverityNumberUnspecified
+	}
+}
+
+// SetIf will modify the given raw map by setting
+// the key and value iff the value is not null and
+// not the empty string.
+func SetIf(attrs map[string]interface{}, key string, value *string) {
+	if value != nil && *value != "" {
+		attrs[key] = *value
+	}
+}
+
+// TraceIDFromGUID will create a pcommon.TraceID from a
+// hex representation of a GUID. All non-hex digits will
+// be stripped from the string before the conversion. The
+// function will return an empty TraceID for any error.
+func TraceIDFromGUID(s *string) pcommon.TraceID {
```

Comment:
```suggestion
func traceIDFromGUID(s *string) pcommon.TraceID {
```

Reply:
I'm not sure if getting a trace ID from the GUID makes sense - could this be peeled off into a separate PR for discussion?

Reply:
All messages/vars have been made private.